ui/admin/app/components/webhooks/WebhookForm.tsx

```diff
@@ -0,0 +1,179 @@
+        <WebhookFormContextProvider {...props}>
+            <WebhookFormContent />
+        </WebhookFormContextProvider>
+    );
+}
+
+export function WebhookFormContent() {
+    const { form, handleSubmit, isLoading, isEdit, hasToken, hasSecret } =
+        useWebhookFormContext();
+
+    const getWorkflows = useSWR(WorkflowService.getWorkflows.key(), () =>
+        WorkflowService.getWorkflows()
+    );
+
+    const workflows = getWorkflows.data;
+
+    // note(ryanhopperlowe): this will change depending on webhook type
+    const validationHeader = form.watch("validationHeader");
+    useEffect(() => {
+        if (!validationHeader) {
+            form.setValue("validationHeader", "X-Hub-Signature-256");
+        }
+    }, [form, validationHeader]);
+
+    return (
+        <form onSubmit={handleSubmit} className="space-y-8 p-8">
+            <TypographyH4>
+                {isEdit ? "Edit Webhook" : "Create Webhook"}
+            </TypographyH4>
+
+            <ControlledInput control={form.control} name="name" label="Name" />
+
+            <ControlledInput
+                control={form.control}
+                name="description"
+                label="Description"
+            />
+
+            <FormItem>
+                <FormLabel>Type</FormLabel>
+                <Select value="Github" disabled>
+                    <SelectTrigger>
+                        <SelectValue />
+                    </SelectTrigger>
+
+                    <SelectContent>
+                        <SelectItem value="Github">Github</SelectItem>
+                    </SelectContent>
+                </Select>
+            </FormItem>
+
+            {/* Extract to custom github component */}
+
+            <ControlledCustomInput
+                control={form.control}
+                name="workflow"
+                label="Workflow"
+                description="The workflow that will be triggered when the webhook is called."
+            >
+                {({ field: { ref: _, ...field }, className }) => (
+                    <Select {...field} onValueChange={field.onChange}>
+                        <SelectTrigger className={className}>
+                            <SelectValue placeholder="Select a workflow" />
+                        </SelectTrigger>
+
+                        <SelectContent>{getWorkflowOptions()}</SelectContent>
+                    </Select>
+                )}
+            </ControlledCustomInput>
+
+            <ControlledInput
+                control={form.control}
+                name="secret"
+                label="Secret"
+                description="This secret should match the secret you provide to GitHub."
+                placeholder={hasSecret ? "(unchanged)" : ""}
+            />
+
+            <ControlledInput
+                control={form.control}
+                name="token"
+                label="Token (optional)"
+                description="Optionally provide a token to add an extra layer of security."
```

Comment:
I think we should either add more information here or link to docs (if they exist) to explain what this does. Whatever token is set will have to be provided by the webhook provider (i.e. GitHub) in a particular query param or header.

Reply:
Ah, I mistakenly assumed the token would be provided in the returned `link` meta. I'll update this

Reply:
I discussed this with Will and he decided to hide the token feature entirely until we can talk it out more with Bill/Donnie